File: bruteforce/6603/6603.java
import java.io.*;
import java.util.*;
import java.lang.*;

public class Main {
    static StringBuilder sb = new StringBuilder();

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		System.out.println("run");;;;

		while (true) {
			String[] arr = br.readLine().split(" ");
			if (arr.length == 1) {
				System.out.println("hit");
				break;
			}
			String[] output = new String[6];
			permutation(arr,output,1,0,0|(0<<1));
			sb.append("\n");
		}

		System.out.println(sb.toString());
	}

	public static void permutation(String[] arr,String[] output, int start,int depth,int flag){
	    if (depth >= 6) {
	         for (int i=0;i<output.length;i++){
	            sb.append(output[i]+" ");
	         }
	         sb.append("\n");
	        return;
	    }else{
			for (int i=start;i<arr.length;i++){ 
				if ((flag & 1<<i) != 0) {
					continue;
				}else{
					output[depth] = arr[i];
					permutation(arr,output,i+1,depth+1,flag | (1<<i));
				}
			}
		}
	    
	}
}

File: bruteforce/6603/6603.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	var (
		reader  = bufio.NewReader(os.Stdin)
		writer  = bufio.NewWriter(os.Stdout)
		builder = strings.Builder{}
	)
	defer writer.Flush()

	for {
		inputs, _ := reader.ReadString('\n')
		input := strings.Split(strings.TrimSpace(inputs), " ")
		if input[0] == "0" {
			fmt.Println(builder.String())
			break
		}

		output := make([]string, 0, 6)
		permutation(&builder, input, output, 1, 0|(0<<1))
		builder.WriteString("\n")
	}
}

func permutation(build *strings.Builder, arr, out []string, start, flag int) {
	if len(out) >= 6 {
		ans := ""
		for _, v := range out {
			ans += v + " "
		}
		build.WriteString(ans + "\n")
		return
	} else {
		for i := start; i < len(arr); i++ {
			if (flag & (1 << i)) != 0 {
				continue
			}
			permutation(build, arr, append(out, arr[i]), i+1, flag|(1<<i))
		}
	}
}
